common/djangoapps/student/models.py

```diff
@@ -1602,6 +1602,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user):
```

Comment:
Also, should this just happen whenever users are created? Maybe in a django signal on post_save for User creation?